shell/components/ResourceDetail/index.vue

```diff
@@ -374,6 +378,8 @@ export default {
       <DetailTop
         v-if="isView && isDetail"
         :value="liveModel"
+        :tooltips="tooltipsDescription"
```

Comment:
`tooltipsDescription` should be a fn in the root `resource-class` that returned undefined (or not and we check for undefined) and have an override in the model for this resource type

```
liveModel?.detailTopToolTip

get detailTopToolTip() { ... }
```

Reply:
This is for the Namespace.
The same function is used also later in the PSA.
We need 2 functions in 2 different models, where we are not needing the class.
Again we need just the utility function here, so there's no gain is moving it to the model.

Reply:
The same explanation given for keeping `DetailTop` generic apply here. There should be no type specific code in generic components. 

I can't see where these generic functions that rely on labels are used outside of namespaces, either here or inhttps://github.com/rancher/dashboard/pull/7678/files. Even if they are used in both they can still either use the same generic function or be in their on `BasePSA` class that the other classes inherit.

Reply:
The three functions getPSALabels, hasPSALabels, getPSATooltipsDescription only apply to namespaces. PodSecurityAdmissionConfigurationTemplate work with a different object.

The description in question here only applies to namespaces.

Keeping resource type code out of generic components is what improves (or does not make worse) the code. There are no other cases where we use resource type code in generic components.

This does not prevent the use of TS or creating simple tests (models are easily created).

Reply:
Moved within model.

Reply:
Resolved by https://github.com/cnotv/dashboard/pull/7